### ISSUE TITLE
serverless login does not work

### ISSUE THREAD

**Issue by franciscocpg** [serverless/serverless#4458]
# This is a Bug Report

## Description

For bug reports:
* What went wrong?
I am trying to run `serverless login` but it doesn't open a tab in my default browser (chrome).

* What did you expect should have happened?
Open a tab in my default browser and continue the login process

* What was the config you used?
* What stacktrace or error message from your provider did you see?

For feature proposals:
* What is the use case that should be solved. The more detail you describe this in the easier it is to understand for us.
* If there is additional config how would it look

Similar or dependent issues:

## Additional Data

* ***Serverless Framework Version you're using***:
1.24.1

* ***Operating System***:
```
Linux 4.13.11-1-ARCH x86_64 GNU/Linux
```
```
NAME="Antergos Linux"
VERSION="17.8-ISO-Rolling"
ID="antergos"
ID_LIKE="arch"
PRETTY_NAME="Antergos Linux"
CPE_NAME="cpe:/o:antergosproject:antergos:17.8"
ANSI_COLOR="1;34;40"
HOME_URL="https://antergos.com/"
SUPPORT_URL="https://forum.antergos.com/"
BUG_REPORT_URL="https://github.com/antergos"
```

I am using `xfce` window manager

* ***Stack Trace***:
```
Serverless: The Serverless login will open in your default browser...
Serverless: Opening browser...
Serverless: Waiting for a successful authentication in the browser.
Serverless: Waiting for a successful authentication in the browser.
Serverless: Waiting for a successful authentication in the browser.
Serverless: Waiting for a successful authentication in the browser.
Serverless: Waiting for a successful authentication in the browser.
Serverless: Waiting for a successful authentication in the browser.
Sorry, something went wrong. Please run "serverless login" again.
 
  Serverless Error ---------------------------------------
 
  Failed to login due to an error. Please try again or contact hello@serverless.com
 
  Stack Trace --------------------------------------------
 
ServerlessError: Failed to login due to an error. Please try again or contact hello@serverless.com
    at getCliLoginById.then.catch (/usr/lib/node_modules/serverless/lib/plugins/login/login.js:167:19)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Forums:        forum.serverless.com
     Chat:          gitter.im/serverless/serverless
 
  Your Environment Information -----------------------------
     OS:                     linux
     Node Version:           8.8.1
     Serverless Version:     1.24.1
```

* ***Provider Error messages***:

**Comment by HyperBrain** [comment]
Hi @franciscocpg ,

I just used the login functionality today and it worked for me (on Windows and on Windows10/Ubuntu shell). Chrome was opened there correctly in both environments.
Maybe the error is to be restricted to Linux (native).

@serverless/vip Did anyone of you encounter this too?

**Comment by franciscocpg** [comment]
hi @HyperBrain 
I took a look at the code and realized two things:
1. serverless is using the `BROWSER` environment variable to open the browser. My `BROWSER` environment variable value was `/usr/bin/chromium` but this path does not exist in my OS. After changing the value to a valid browser path in my OS the problem was solved.
2. There is a bug in the code handling this kind of error. So I've opened https://github.com/serverless/serverless/pull/4460 to fix it.
